src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -488,4 +442,33 @@ private void registerRetryConnectionBroadcastReceiver() {
 		};
 		registerReceiver(broadcastReceiver, new IntentFilter("retryConnection"));
 	}
+
+//> CLASSES
+	public static class RequestCode {
```

Comment:
Minor, but this looks like a pretty good place to use a Java enum:

```.java
	public enum RequestCode {
		ACCESS_LOCATION_PERMISSION(100),
		ACCESS_STORAGE_PERMISSION(101),
		CHT_EXTERNAL_APP_ACTIVITY(102),
		DISCLOSURE_LOCATION_ACTIVITY(103),
		GRAB_MRDT_PHOTO_ACTIVITY(104),
		GRAB_PHOTO_ACTIVITY(105);

		private final int requestCd;

		RequestCode(int requestCd) {
			this.requestCd = requestCd;
		}

		public static Optional<RequestCode> valueOf(int requestCd) {
			return Arrays.stream(RequestCode.values()).filter(e -> e.getCode() == requestCd).findFirst();
		}

		public int getCode() {
			return requestCd;
		}
	}
```

Then above you can just switch directly on the enum itself.  (Also, if you need the string value to trace out, you can just use the `RequestCode.name()` method.)

Reply:
Nice suggestion! It's done